homeassistant/components/hausbus/light.py

```diff
@@ -0,0 +1,274 @@
+"""Support for Haus-Bus lights."""
+import colorsys
+from typing import Any, cast
+
+from pyhausbus.ABusFeature import ABusFeature
+from pyhausbus.de.hausbus.homeassistant.proxy.Dimmer import Dimmer
+from pyhausbus.de.hausbus.homeassistant.proxy.dimmer.data.EvOff import (
+    EvOff as DimmerEvOff,
+)
+from pyhausbus.de.hausbus.homeassistant.proxy.dimmer.data.EvOn import EvOn as DimmerEvOn
+from pyhausbus.de.hausbus.homeassistant.proxy.dimmer.data.Status import (
+    Status as DimmerStatus,
+)
+from pyhausbus.de.hausbus.homeassistant.proxy.Led import Led
+from pyhausbus.de.hausbus.homeassistant.proxy.led.data.EvOff import EvOff as ledEvOff
+from pyhausbus.de.hausbus.homeassistant.proxy.led.data.EvOn import EvOn as ledEvOn
+from pyhausbus.de.hausbus.homeassistant.proxy.led.data.Status import Status as ledStatus
+from pyhausbus.de.hausbus.homeassistant.proxy.RGBDimmer import RGBDimmer
+from pyhausbus.de.hausbus.homeassistant.proxy.rGBDimmer.data.EvOff import (
+    EvOff as rgbDimmerEvOff,
+)
+from pyhausbus.de.hausbus.homeassistant.proxy.rGBDimmer.data.EvOn import (
+    EvOn as rgbDimmerEvOn,
+)
+from pyhausbus.de.hausbus.homeassistant.proxy.rGBDimmer.data.Status import (
+    Status as rgbDimmerStatus,
+)
+
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS,
+    ATTR_HS_COLOR,
+    DOMAIN,
+    ColorMode,
+    LightEntity,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .channel import HausbusChannel
+from .const import ATTR_ON_STATE, DOMAIN as HAUSBUSDOMAIN
+from .device import HausbusDevice
+from .event_handler import IEventHandler
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Haus-Bus lights from a config entry."""
+    gateway = cast(IEventHandler, hass.data[HAUSBUSDOMAIN][config_entry.entry_id])
+
+    @callback
+    async def async_add_light(channel: HausbusChannel) -> None:
+        """Add light from Haus-Bus."""
+        entities: list[HausbusLight] = []
+        if isinstance(channel, HausbusLight):
+            entities.append(channel)
+        async_add_entities(entities)
+
+    gateway.register_platform_add_channel_callback(async_add_light, DOMAIN)
+
+
+class HausbusLight(HausbusChannel, LightEntity):
```

Comment:
if you split this into a `HausBusLight` and from that inherit then `HausBusDimmerLight`, `HausBusLEDLight`, `HausBusRGBLight` classes you can avoid all those `if`s, because you have a specific class per type

Reply:
This was easier than expected, thanks.